src/stirling/source_connectors/socket_tracer/thriftmux_openssl_trace_bpf_test.cc

```diff
@@ -0,0 +1,177 @@
+bool Init() {
+  // Make sure Mux tracing is enabled.
+  FLAGS_stirling_enable_mux_tracing = true;
+
+  // Enable the raw fptr fallback for determining ssl lib version.
+  FLAGS_openssl_raw_fptrs_enabled = true;
+  return true;
+}
+
+bool kInit = Init();
+
+template <typename TServerContainer>
+class BaseOpenSSLTraceTest : public SocketTraceBPFTestFixture</* TClientSideTracing */ false> {
+ protected:
+  BaseOpenSSLTraceTest() {
+    // Run the nginx HTTPS server.
+    // The container runner will make sure it is in the ready state before unblocking.
+    // Stirling will run after this unblocks, as part of SocketTraceBPFTestFixture SetUp().
+    StatusOr<std::string> run_result =
+        server_.Run(std::chrono::seconds{60}, {}, {"--use-tls", "true"});
+    PL_CHECK_OK(run_result);
+    PL_CHECK_OK(this->RunThriftMuxClient());
+
+    // Sleep an additional second, just to be safe.
+    sleep(1);
+  }
+
+  StatusOr<int32_t> RunThriftMuxClient() {
+    std::string classpath =
+        "@/app/px/src/stirling/source_connectors/socket_tracer/testing/containers/thriftmux/"
+        "server_image.classpath";
+    std::string thriftmux_client_output = "StringString";
+
+    // The thriftmux container uses a CA created by
+    // src/common/testing/test_utils/cert_generator:cert_generator.
+    // This command adds this CA to java's keystore so ssl
+    // verification works as expected.
+    std::string keytool_cmd = absl::Substitute(
+        "docker exec $0 /usr/lib/jvm/java-11-openjdk-amd64/bin/keytool -importcert -keystore "
+        "/etc/ssl/certs/java/cacerts -file /etc/ssl/ca.crt -noprompt -storepass changeit",
+        server_.container_name());
+    px::Exec(keytool_cmd);
+
+    // Runs the Client entrypoint inside the server container
+    // capturing the Client process's pid for easier debugging.
+    std::string cmd = absl::Substitute(
+        "docker exec $0 /usr/bin/java -cp $1 Client --use-tls true & echo $$! && wait",
+        server_.container_name(), classpath);
+    PL_ASSIGN_OR_RETURN(std::string out, px::Exec(cmd));
+
+    LOG(INFO) << absl::Substitute("thriftmux client command output: '$0'", out);
+
+    std::vector<std::string> tokens = absl::StrSplit(out, "\n");
+
+    int32_t client_pid;
+    if (!absl::SimpleAtoi(tokens[0], &client_pid)) {
+      return error::Internal("Could not extract PID.");
+    }
+
+    LOG(INFO) << absl::Substitute("Client PID: $0", client_pid);
+
+    if (!absl::StrContains(out, thriftmux_client_output)) {
+      return error::Internal(
+          "Expected output from thriftmux to include '$0', received '$1' instead",
+          thriftmux_client_output, out);
+    }
+    return client_pid;
+  }
+
+  TServerContainer server_;
+};
+
+mux::Record RecordWithType(mux::Type req_type) {
+  mux::Record r = {};
+  r.req.type = static_cast<int8_t>(req_type);
+
+  return r;
+}
+
+using ThriftMuxTLSServerImplementations = ::testing::Types<ThriftMuxServerContainerWrapper>;
```

Comment:
Do we have any other servers we expect to run in the future? If not, we should take out the templates and simplify this. Thoughts?

Reply:
Other netty integrations could take advantage of this test assuming our expectation that they will work out of the box is correct. Maybe we should name this `netty_{ssl,tls}_trace_bpf_test.cc` instead?

Reply:
I chose `netty_tls_trace_bpf_test.cc` since I felt that not including "tls" part of the name was important to reflect and agree that it might be good to further consolidate to something "boringssl" related.